pkg/controller/v1alpha2/applicationconfiguration/dependency_test.go

```diff
@@ -517,17 +504,13 @@ var _ = Describe("Resource Dependency in an ApplicationConfiguration", func() {
 		}, time.Second, 300*time.Millisecond).Should(BeTrue())
 
 		By("Reconcile")
-		Expect(func() error { _, err := reconciler.Reconcile(req); return err }()).Should(BeNil())
+		reconcileRetry(reconciler, req)
 
 		By("Verify the appconfig's dependency is satisfied")
 		Eventually(
 			func() []v1alpha2.UnstaifiedDependency {
 				tempAppConfig := &v1alpha2.ApplicationConfiguration{}
-				err := k8sClient.Get(ctx, appconfigKey, tempAppConfig)
-				if err != nil || tempAppConfig.Status.Dependency.Unsatisfied != nil {
-					// Try 3 (= 1s/300ms) times
-					reconciler.Reconcile(req)
```

Comment:
why not keep the reconcile here？

Reply:
Got it, I will revert the change here. It's diff from other reconcile logic.